source/upgrading/upgrade/upgrade-4.22.rst

```diff
@@ -0,0 +1,309 @@
+
+   .. parsed-literal::
+
+      $ sudo apt-get update
+
+#. Now that you have the repository configured, it's time to upgrade
+   the ``cloudstack-management`` package.
+
+   .. parsed-literal::
+
+      $ sudo apt-get install cloudstack-management
+
+#. If you use CloudStack usage server
+
+   .. parsed-literal::
+
+      $ sudo apt-get install cloudstack-usage
+
+
+.. _rhel414:
+.. _rpm-repo414:
+
+CentOS/RHEL
+##############
+
+If you are using CentOS or RHEL, follow this procedure to upgrade your
+packages. If not, skip to hypervisors section :ref:`upg_hyp_414`.
+
+.. note::
+   **Community Packages:** This section assumes you're using the community
+   supplied packages for CloudStack. If you've created your own packages and
+   yum repository, substitute your own URL for the ones used in these examples.
+
+The first order of business will be to change the yum repository
+for each system with CloudStack packages. This means all
+management servers, and any hosts that have the KVM agent (no changes
+should be necessary for hosts that are running VMware or Xen.)
+
+Change your ``/etc/yum.repos.d/cloudstack.repo`` file on
+any systems that have CloudStack packages installed to points to version |version|.
+
+This file should have content similar to the following:
+
+.. parsed-literal::
+
+   [apache-cloudstack]
+   name=Apache CloudStack
+   baseurl=http://download.cloudstack.org/centos/$releasever/|version|/
+   enabled=1
+   gpgcheck=0
+
+Setup the GPG public key if you wish to enable ``gpgcheck=1``:
+
+.. parsed-literal::
+
+   rpm --import http://download.cloudstack.org/RPM-GPG-KEY
+
+#. Now that you have the repository configured, it's time to upgrade the
+   ``cloudstack-management``.
+
+   .. parsed-literal::
+
+      $ sudo yum upgrade cloudstack-management
+
+#. If you use CloudStack usage server
+
+   .. parsed-literal::
+
+      $ sudo yum upgrade cloudstack-usage
+
+.. _upg_hyp_414:
+
+Upgrade Hypervisors
+-------------------
+
+Hypervisor: XenServer
+#####################
+
+
+No additional steps are required for XenServer Hypervisor for this upgrade.
+
+
+Hypervisor: VMware
+###################
+
+.. warning::
+   For VMware hypervisor, CloudStack management server packages must be
+   built using "noredist". Refer to :ref:`building-noredist`.
+
+
+No additional steps are required for the VMware Hypervisor for this upgrade.
+
+
+.. _kvm414:
+
+Hypervisor: KVM
+#################
+
+KVM on Ubuntu/Debian
+""""""""""""""
+
+(KVM only) Additional steps are required for each KVM host. These
+steps will not affect running guests in the cloud. These steps are
+required only for clouds using KVM as hosts and only on the KVM
+hosts.
+
+#. Configure the :ref:`APT repo <apt-repo414>` as detailed above.
+
+#. Stop the running agent.
+
+   .. parsed-literal::
+
+      $ sudo systemctl stop cloudstack-agent
+
+#. Update the agent software.
+
+   .. parsed-literal::
+
+      $ sudo apt-get upgrade cloudstack-agent
```

Comment:
install / upgrade? - keep in sync with _cloudstack-management, cloudstack-usage_